src/Compilers/Core/Portable/Text/CompositeText.cs

```diff
@@ -373,5 +378,125 @@ private static void TrimInaccessibleText(ArrayBuilder<SourceText> segments)
                 segments.Add(writer.ToSourceText());
             }
         }
+
+        /// <summary>
+        /// Delegates to SourceTexts within the CompositeText to determine line information.
+        /// </summary>
+        internal sealed class CompositeLineInfo : TextLineCollection
+        {
+            private readonly CompositeText _compositeText;
+            private readonly int[] _segmentLineIndexes;
+            private readonly int _lineCount;
```

Comment:
1. maybe doc tehse.  i'd like docs to understand how lineCount and _segmentLineIndexes relates.
2. 'indices' instead of 'indexes'?

Reply:
Renamed and doc'ed, hopefully clearer now